TradeItIosTicketSDK2/TradeItAlertManager.swift

```diff
@@ -4,67 +4,65 @@ import UIKit
     var linkedBrokerManager = TradeItLauncher.linkedBrokerManager
     var linkBrokerUIFlow = TradeItLinkBrokerUIFlow(linkedBrokerManager: TradeItLauncher.linkedBrokerManager)
 
-    func showGenericError(tradeItErrorResult tradeItErrorResult: TradeItErrorResult,
-                            onViewController viewController: UIViewController,
-                                             onFinished: () -> Void = {}) {
-        let alertTitle = tradeItErrorResult.shortMessage ?? ""
-        let messages = (tradeItErrorResult.longMessages as? [String]) ?? []
-        let alertMessage = messages.joinWithSeparator(" ")
-        let alertActionTitle = "OK"
+    public func showError(error: TradeItErrorResult,
+                          onViewController viewController: UIViewController,
+                          onFinished: () -> Void = {}) {
+        let title = error.shortMessage ?? ""
+        let messages = (error.longMessages as? [String]) ?? []
+        let message = messages.joinWithSeparator(". ")
+        let actionTitle = "OK"
 
-        self.showOn(viewController: viewController,
-                    withAlertTitle: alertTitle,
-                    withAlertMessage: alertMessage,
-                    withAlertActionTitle: alertActionTitle,
+        self.showAlert(onViewController: viewController,
+                              withTitle: title,
+                            withMessage: message,
+                        withActionTitle: actionTitle,
                     onAlertActionTapped: onFinished)
     }
 
-    public func show(tradeItErrorResult tradeItErrorResult: TradeItErrorResult,
-                onViewController viewController: UIViewController,
-                withLinkedBroker linkedBroker: TradeItLinkedBroker,
-                                 onFinished : () -> Void) {
+    public func showRelinkError(error: TradeItErrorResult,
```

Comment:
This feels more like a "recover from error if possible" helper. The user would have to implement the logic for checking the type of error otherwise which feels unnecessary to me.